### PR TITLE
Update networkmgr.lua

### PR DESCRIPTION
**Pull request by Markismus**
Change from the commands from `wifi.sh` to those from `wifi_enable_dhcp.sh` from Tshering's Start Menu. See Issue #939. Should resolve hanging of Kobo Aura H2O.

### CODE REVIEW

**Comment by NiLuJe** [comment]
Do we always inherit an env with PLATFORM set?

**Comment by Frenzie** [comment]
Presumably only when started from (a sufficiently recent version of?) Kobo Start Menu. 

A suboptimal solution would be to prefix the stuff with something like

```
if [ -n "$PLATFORM" ] PLATFORM='ntx508'
```

**Comment by Markismus** [comment]
I've assumed it, because it's defined in rcS. Both alternatives in [#939](https://github.com/koreader/koreader/issues/939#issuecomment-73420146) reacted the same, so that is an extra argument for it. 
However, it would be nice to see what is dumped to standard out when toggling WiFi on.

**Comment by Markismus** [comment]
@Frenzie No I think it is dependent on rcS. Tshering puts his modifications between comments, so this should be in any newer firmware.

**Comment by NiLuJe** [comment]
Se my previous edit, it's in the default rcS, but only there. Which means only stuff started from within rcS will inherit it. That's... suboptimal in terms of portability ;). (It breaks on telnet/ssh session, for instance).

**Comment by Markismus** [comment]
I would rather include the commands from rcS. Maybe a prefix 

```
if [ -n "$PLATFORM" ]; then 
  PLATFORM=freescale
  if [ `dd if=/dev/mmcblk0 bs=512 skip=1024 count=1 | grep -c "HW CONFIG"` == 1 ]; then
    CPU=`ntx_hwconfig -s -p /dev/mmcblk0 CPU`
    PLATFORM=$CPU-ntx
  fi

  if [ $PLATFORM == freescale ]; then
    if [ ! -s /lib/firmware/imx/epdc_E60_V220.fw ]; then
      mkdir -p /lib/firmware/imx
      dd if=/dev/mmcblk0 bs=512K skip=10 count=1 | zcat > /lib/firmware/imx/epdc_E60_V220.fw
      sync
    fi
  elif [ ! -e /etc/u-boot/$PLATFORM/u-boot.mmc ]; then
    PLATFORM=ntx508
  fi
fi
```

How can you write a multi-line execute.os statement?

**Comment by NiLuJe** [comment]
Put it all in a shell script, ship it, and execute that instead?

**Comment by Markismus** [comment]
I remember that pazos moved it into a lua-script, because of the wild growth of kobo tailored scripts. I don't mind, whatever works faster. That's why the question for [multi-line commands](http://stackoverflow.com/questions/13077241/execute-combine-multiple-linux-commands-in-one-line)

**Comment by Frenzie** [comment]
If it takes a regular string then Lua uses double brackets, does it not?

```lua
[[line 1
line 2
line 3]]
```

File: frontend/ui/networkmgr.lua [hunk 1]
```diff
@@ -21,14 +21,14 @@ end
...
+        os.execute("lsmod | grep -q sdio_wifi_pwr || insmod /drivers/$PLATFORM/wifi/sdio_wifi_pwr.ko")
+        os.execute("lsmod | grep -q dhd || insmod /drivers/$PLATFORM/wifi/dhd.ko")
+        os.execute("sleep 2")
         os.execute("ifconfig eth0 up")
         os.execute("wlarm_le -i eth0 up")
-        os.execute("wpa_supplicant -s -i eth0 -c "..path.." -C /var/run/wpa_supplicant -B")
-        os.execute("udhcpc -S -i eth0 -s /etc/udhcpc.d/default.script -t15 -T10 -A3 -b -q >/dev/null 2>&1")
+        os.execute("pidof wpa_supplicant >/dev/null || wpa_supplicant -s -i eth0 -c /etc/wpa_supplicant/wpa_supplicant.conf -C /var/run/wpa_supplicant -B")
+        os.execute("sleep 1")
+        os.execute("sbin/udhcpc -S -i eth0 -s /etc/udhcpc.d/default.script -t15 -T10 -A3 -b -q >/dev/null 2>&1 &")
```
Comment (chrox) [inline frontend/ui/networkmgr.lua]: Is this `sbin/udhcpc` or `/sbin/udhcpc`? I'm afraid current koreader directory does not have the `sbin/udhcpc`.

Comment (houqp) [inline frontend/ui/networkmgr.lua]: good catch, it must have been broken for a long time :(

Comment (Markismus) [inline frontend/ui/networkmgr.lua]: It is indeed `/sbin/udhcpc`. Typo. I checked twice! Thanks.

**Comment by Markismus** [comment]
Alright. The typo is corrected and if `PLATFORM` is undefined than the value will be assigned in `koreader.sh`.

**Pull request merged**